CarRentalPoint/CarRentalPoint.Tests/CarRentalServiceTests.cs

```diff
@@ -0,0 +1,223 @@
+public class CarRentalServiceTests(CarRentalDataSeeder Service) : IClassFixture<CarRentalDataSeeder>
+{
+    /// <summary>
+    /// Tests that clients who rented a specific car model are returned and ordered by full name
+    /// Verifies the correct filtering and sorting logic
+    /// </summary>
+    [Fact]
+    public void GetClientsByCarModel_ShouldReturnClientsOrderedByFullName()
+    {
+        // Arrange
+        var targetModelId = 1;
+        var expectedCount = 8;
+
+        // Act
+        var result = Service.Rentals
+            .Where(r => r.Car.Generation.Model.Id == targetModelId)
+            .Select(r => r.Client)
+            .Distinct()
+            .OrderBy(c => c.FullName)
+            .ToList();
+
+        // Assert
+        Assert.Equal(expectedCount, result.Count);
+    }
+
+    /// <summary>
+    /// Tests retrieval of currently rented cars using fixed dates
+    /// Verifies that only cars with active rental periods are returned
+    /// </summary>
+    [Fact]
+    public void GetCarsCurrentlyRented_ShouldReturnActiveRentals()
+    {
+        // Arrange
+        var rentals = Service.Rentals;
+        var currentTime = new DateTime(2024, 1, 15, 14, 0, 0); // Фиксированное время для теста
+        var expectedCount = 3;
+
+        // Act
+        var activeRentals = rentals
+            .Where(r => IsRentalActive(r, currentTime))
+            .Select(r => r.Car)
+            .Distinct()
+            .ToList();
+
+        // Assert
+        Assert.Equal(expectedCount, activeRentals.Count);
+    }
+
+    /// <summary>
+    /// Helper method to determine if rental is active at specific time
+    /// </summary>
+    private bool IsRentalActive(Rental rental, DateTime currentTime)
+    {
+        var rentalEnd = rental.RentalStart.AddHours(rental.RentalHours);
+        return rental.RentalStart <= currentTime && currentTime <= rentalEnd;
+    }
+
+    /// <summary>
+    /// Tests the retrieval of top 5 most frequently rented cars
+    /// Verifies correct ordering by rental count in descending order
+    /// </summary>
+    [Fact]
+    public void GetTop5MostFrequentlyRentedCars_ShouldReturnExpectedCars()
+    {
+        // Arrange
+        var rentals = Service.Rentals;
+        var expectedTopCount = 5;
+
+        // Ожидаемое множество машин (например, заранее знаем из сидера, что чаще всего брали вот эти)
+        var expectedCars = new List<Car>
+        {
+            Service.Cars[0],  // Id = 1 (5 аренд)
+            Service.Cars[3],  // Id = 4 (3 аренды)
+            Service.Cars[5],  // Id = 6 (3 аренды)
+            Service.Cars[7],  // Id = 8 (2 аренды)
+            Service.Cars[10]  // Id = 11 (2 аренды)
+        };
+
+
+        // Act
+        var topCars = rentals
+            .GroupBy(r => r.Car.Id)
+            .Select(g => new
+             {
+                 CarId = g.Key,
+                 RentalCount = g.Count(),
+                 Car = g.First().Car  // Берем первый объект Car из группы
+             })
```

Comment:
Можно вот так сделать, если название справа совпадает с нужным названием
```suggestion
            .Select(g => new
             {
                 CarId = g.Key,
                 RentalCount = g.Count(),
                 g.First().Car 
             })
```

Ниже есть такие же места, там тоже поправить